### PR TITLE
coprocessor/expression: Add ToSeconds/Quarter function

### PR DESCRIPTION
**Pull request by JaySon-Huang**
## What have you changed? (mandatory)

- Add `to_seconds`, `quarter` function in coprocessor with corresponding unit tests.

## What are the type of the changes? (mandatory)

- New feature (change which adds functionality)

## How has this PR been tested? (mandatory)

- unit test

## Does this PR affect documentation (docs) or release note? (mandatory)

- no

## Does this PR affect tidb-ansible update? (mandatory)

- no

## Refer to a related PR or issue link (optional)

#3275

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
I rebase the code to the newest tikv-master then do a force push to my repo, then I found this PR closed. Reopen this PR.

**Comment by breeswish** [comment]
Sorry we missed this PR. Could you merge with latest master to resolve conflicts?

**Comment by JaySon-Huang** [comment]
@breeswish PTAL

**Comment by breeswish** [comment]
@iosmanthus PTAL, thanks!

File: components/tidb_query/src/expr/builtin_time.rs [hunk 1]
```diff
@@ -315,6 +315,28 @@ impl ScalarFunc {
         Ok(Some(i64::from(t.day_number())))
     }
 
+    #[inline]
+    pub fn to_seconds(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let t: Cow<'_, Time> = try_opt!(self.children[0].eval_time(ctx, row));
+        if t.is_zero() {
+            return ctx
+                .handle_invalid_time_error(Error::incorrect_datetime_value(&format!("{}", t)))
```
Comment (iosmanthus) [inline components/tidb_query/src/expr/builtin_time.rs]: `Error::incorrect_datetime_value(t)`

Comment (iosmanthus) [reply components/tidb_query/src/expr/builtin_time.rs]: Rest LGTM

Comment (JaySon-Huang) [reply components/tidb_query/src/expr/builtin_time.rs]: done

**Comment by JaySon-Huang** [comment]
@iosmanthus PTAL

**Comment by breeswish** [comment]
Thanks for the contribution! Would you like to participate in our [PCP](https://github.com/pingcap/community/tree/master/challenge-Programs) so that you can get rewarded after merging this PR?

**Comment by JaySon-Huang** [comment]
Sign up done @breeswish . https://github.com/tidb-perf-challenge/pcp/issues/112

**Comment by zz-jason** [comment]
@JaySon-Huang There hasn't been any update for more than 1 year. I'm going to close this stale PR, feel free to reopen it once you wish to solve the issue in the future.

**Pull request closed without merging**